diff --git cef.gyp cef.gyp
index e25eff3..77b8e11 100644
--- cef.gyp
+++ cef.gyp
@@ -1742,7 +1742,10 @@
       'target_defaults': {
         # Disable warnings about c++0x compatibility, as some names (such
         # as nullptr) conflict with upcoming c++0x types.
-        'cflags_cc': ['-Wno-c++0x-compat'],
+        'cflags_cc': [
+            '-Wno-c++0x-compat',
+            '-Wno-literal-suffix'
+        ],
       },
     }],
     ['OS=="mac"', {
