diff --git ui/views/widget/desktop_aura/desktop_window_tree_host_x11.cc ui/views/widget/desktop_aura/desktop_window_tree_host_x11.cc
index b9e80b6..65a32eb 100644
--- ui/views/widget/desktop_aura/desktop_window_tree_host_x11.cc
+++ ui/views/widget/desktop_aura/desktop_window_tree_host_x11.cc
@@ -176,8 +176,8 @@ DesktopWindowTreeHostX11::DesktopWindowTreeHostX11(
       custom_window_shape_(false),
       urgency_hint_set_(false),
       activatable_(true),
-      close_widget_factory_(this),
-      xwindow_destroyed_(false) {
+      xwindow_destroyed_(false),
+      close_widget_factory_(this) {
 }
 
 DesktopWindowTreeHostX11::~DesktopWindowTreeHostX11() {
diff --git ui/views/widget/desktop_aura/desktop_window_tree_host_x11.h ui/views/widget/desktop_aura/desktop_window_tree_host_x11.h
index bc5376f..e1b6bc3 100644
--- ui/views/widget/desktop_aura/desktop_window_tree_host_x11.h
+++ ui/views/widget/desktop_aura/desktop_window_tree_host_x11.h
@@ -361,11 +361,11 @@ class VIEWS_EXPORT DesktopWindowTreeHostX11
 
   base::CancelableCallback<void()> delayed_resize_task_;
 
-  base::WeakPtrFactory<DesktopWindowTreeHostX11> close_widget_factory_;
-
   // True if the xwindow has already been destroyed.
   bool xwindow_destroyed_;
 
+  base::WeakPtrFactory<DesktopWindowTreeHostX11> close_widget_factory_;
+
   DISALLOW_COPY_AND_ASSIGN(DesktopWindowTreeHostX11);
 };
 
