diff --git cef_atomicops_arm_gcc.h cef_atomicops_arm_gcc.h
index e187bee..9fb1511
--- cef_atomicops_arm_gcc.h
+++ cef_atomicops_arm_gcc.h
@@ -63,7 +63,8 @@ inline void MemoryBarrier() {
     defined(__ARM_ARCH_7R__) || defined(__ARM_ARCH_7M__) || \
     defined(__ARM_ARCH_6__) || defined(__ARM_ARCH_6J__) || \
     defined(__ARM_ARCH_6K__) || defined(__ARM_ARCH_6Z__) || \
-    defined(__ARM_ARCH_6ZK__) || defined(__ARM_ARCH_6T2__)
+    defined(__ARM_ARCH_6ZK__) || defined(__ARM_ARCH_6T2__) || \
+    defined(__ARM_ARCH_8A__)
 
 inline Atomic32 NoBarrier_CompareAndSwap(volatile Atomic32* ptr,
                                          Atomic32 old_value,
